### PR TITLE
Rbotha/feature 167 view hazards detail hazard card elements

### PR DESCRIPTION
**Pull request by ReinhardtBotha**
- Moved MyReportCard and HazardDetailCard sections into the ReportCardContainer class
- MyReportCard and HazardDetailCard now calls the methods from the class 
- Hide don't show gallery and details sections in cards if they don't have data
- Border lines now added using pseudo classes
- fixed some styling 

fix https://github.com/whamester/WMDD-4885-Project/issues/85

### CODE REVIEW

File: assets/components/ReportCardContainer.js [hunk 1]
```diff
@@ -33,17 +34,196 @@ class ReportCardContainer {
...
+    div.setAttribute('class', 'report-card__top-controls');
+    div.innerHTML = contentHTML;
+    return div;
+  }
+
+  getHeading() {
+    const icon = this.settings?.detail;
+    const contentHTML = `
+			<span class="btn__icon report-card__heading__icon" style="background-color: ${icon.iconBackround}">
+				<i class="${icon.icon}" style="width:1.5rem; height:1.5rem; background-color: white"></i>
```
Comment (whamester) [inline assets/components/ReportCardContainer.js:60]: var(--white)

File: assets/components/ReportCardContainer.js [hunk 2]
```diff
@@ -33,17 +34,196 @@ class ReportCardContainer {
...
+  
+    <div id="flagReportMessage" class="message error">
+      <i class="icon-flag message__icon"></i>
+      <p class="message__content text-body-3 medium">
+        You have flagged this report as fake.
+      </p>
+    </div>
+    `;
+    const div = document.createElement('div');
+    div.setAttribute('class', "report-card__hazard-detail-buttons ${this.createdByUserLoggedIn ? 'hidden' : ''}");
```
Comment (whamester) [inline assets/components/ReportCardContainer.js:224]: Does this need to be inside backticks?
```
"report-card__hazard-detail-buttons ${this.createdByUserLoggedIn ? 'hidden' : ''}"
```

File: assets/components/ReportCardContainer.js (hunk 1 above)
Comment (ReinhardtBotha) [reply assets/components/ReportCardContainer.js]: addressed in: https://github.com/whamester/WMDD-4885-Project/pull/122/commits/6c47bbcbff6ca76a64e9ecb9aa9e812b90ca0a40

File: assets/components/ReportCardContainer.js (hunk 2 above)
Comment (ReinhardtBotha) [reply assets/components/ReportCardContainer.js]: thanks! https://github.com/whamester/WMDD-4885-Project/pull/122/commits/6c47bbcbff6ca76a64e9ecb9aa9e812b90ca0a40

**Review by whamester — APPROVED** [review]

**Pull request merged**